### ISSUE TITLE
Rethink how `null` values are handled

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/accessible-autocomplete#260]
User should be able to submit a form with autocomplete with a `null` value because they're being presented with a text box (the UI itself doesn't suggest it is a select) and text boxes natively allow you to submit a null value on the client side (thanks @joelanman!).

We should add to guidance that `select` should always have `<option value=""></option>` to ensure that even if user is getting the native `select` experience, they can still submit a `null` value.

Handling of a `null` value should be the concern of server side validation.

We should consider injecting a null value into a select if we detect it doesn't contain one.

We could also consider adding some optional client side validation (eg. if user opens autocomplete but doesn't select a value, show error message)

The guidance here would need to be rethought: https://github.com/alphagov/accessible-autocomplete#null-options

Linked to:
https://github.com/alphagov/accessible-autocomplete/issues/220
https://github.com/alphagov/accessible-autocomplete/issues/236

**Comment by HughePaul** [comment]
for a select box with enhancement: If text was entered that didn't match any results and that was submitted, i'd expect the field to behave as if no item had been selected, and normal form validation should handle that in the same way it would for a normal select box.

**Comment by edwardhorsford** [comment]
Although the autocomplete progressively enhances a select, it could replace it with a input - which is what it looks like anyway. Then the input can submit whatever inputs are allowed to submit. If the response has to be one of a set values, the autocomplete can already handle clearing entries that don't match.

I don't see why we'd need to have selects with null values - the only reason we have it is because that's how inputs work. If it doesn't look like an input we don't need to make it behave like one.

**Comment by HughePaul** [comment]
submitting the non-matching query value (as a text input) would be better than submitting a previously selected (or default) item that the user can no longer see. then server side validation could complain.

**Comment by nubz** [comment]
Yes I agree @HughePaul - clearing the input gives the illusion to the user that you have now emptied it, but we can see under the hood the select still has the previously selected item still selected and going into the payload - this is particularly important on optional fields where an empty string is a valid response, it is not possible to select an empty string after selecting a matching option without fixing this bug on handleInputChange which simply sets the state to options: [] for exmpty queries and does not reset the select element to it's default value (which I think is the expected behaviour)
